langchain/document_loaders/pdf.py

```diff
@@ -381,3 +381,91 @@ def load(self) -> List[Document]:
             contents = self.clean_pdf(contents)
         metadata = {"source": self.source, "file_path": self.source}
         return [Document(page_content=contents, metadata=metadata)]
+
+
+class PDFPlumberLoader(BasePDFLoader):
+    """Loader that uses PDFPlumber to load PDF files."""
+
+    def __init__(self, file_path: str):
+        """Initialize with file path."""
+        try:
+            import pdfplumber  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "PDFPlumber package not found, please install it with "
+                "`pip install pdfplumber`"
+            )
+
+        super().__init__(file_path)
+
+    def load(self, **kwargs: Optional[Any]) -> List[Document]:
+        """Load file."""
+        import pdfplumber
+
+        doc = pdfplumber.open(self.file_path)
+        file_path = self.file_path if self.web_path is None else self.web_path
+
+        return [
+            Document(
+                page_content=page.extract_text(**kwargs).encode("utf-8"),
+                metadata=dict(
+                    {
+                        "source": file_path,
+                        "file_path": file_path,
+                        "page_number": page.page_number,
+                        "total_pages": len(doc.pages),
+                    },
+                    **{
+                        k: doc.metadata[k]
+                        for k in doc.metadata
+                        if type(doc.metadata[k]) in [str, int]
+                    },
+                ),
+            )
+            for page in doc.pages
+        ]
+
+    def annotate_and_load(
+        self, folder_path: str, **kwargs: Optional[Any]
```

Comment:
small thing, could we give folder_path a slightly more descriptive name (like `save_path` or `image_path`)

Reply:
also kwargs can just be typed as Any, no need for Optional

Reply:
Fixed. `folder_path` is the naming scheme for the saving location of the FAISS `save_local` method, maybe we should rename that as well?